sys/sys/systm.h

```diff
@@ -122,18 +122,20 @@ struct ucred;
  * Derive out-of-bounds and small values from NULL.  This allows common
  * sentinel values to work.
  */
-#define ___USER_CFROMPTR(ptr, cap)					\
+#define ___USER_CFROMPTR(ptr, cap, is_offset)				\
     ((void *)(uintptr_t)(ptr) == NULL ? NULL :				\
      ((vm_offset_t)(ptr) < 4096 ||					\
       (vm_offset_t)(ptr) > VM_MAXUSER_ADDRESS) ?			\
-	__builtin_cheri_offset_set(NULL, (ptraddr_t)(ptr)) :		\
-	__builtin_cheri_offset_set((cap), (ptraddr_t)(ptr)))
+	__builtin_cheri_address_set(NULL, (ptraddr_t)(ptr)) :		\
```

Comment:
`(void * __capability)(uintcap_t)(ptraddr_t)(ptr)` might be a less-weird way to write this these days that doesn't need to pick one of address and offset, FWIW

Reply:
https://github.com/CTSRD-CHERI/cheribsd/pull/1635